api/room/schema_query.py

```diff
@@ -367,13 +370,19 @@ def resolve_analytics_ratios(
             self, query, start_date, end_date)
         return ratio
 
-    @Auth.user_roles('Admin', 'Default User')
+    @Auth.user_roles('Admin')
     def resolve_analytics_ratios_per_room(
-            self, info, start_date, end_date=None):
+            self, info, start_date, end_date=None, room_id=None):
         query = Room.get_query(info)
+        if room_id:
+            check_room = query.filter(RoomModel.id == room_id).first()
+            if not check_room:
+                raise GraphQLError("Room not found")
         ratio = RoomAnalyticsRatios.get_analytics_ratios_per_room(
-            self, query, start_date, end_date)
-        return RatiosPerRoom(ratio)
+            self, query, start_date, end_date, room_id)
+        if room_id:
```

Comment:
Hello @jesseinit , great work. Is it possible to away with this condition here and work with only the first condition in line 377?

Reply:
The rationale behind using the second `if` conditional was an attempt to keep the implementation DRY and most importantly specify what property in `RatiosPerRoom` class we are returning the resultset to.
I'd be glad if you can proffer a better approach to solving this.

Reply:
Hello @jesseinit , I have reviewed this implementation.  this endpoint exists on develop but not working as it is meant to be. So the refactor can help here. You do not need to check if room id is provided because the method takes in a room_id. All you need to do is to verify that that room_id exists.  You can do something like this below.
```
def resolve_analytics_ratios_per_room(self,
                                          info,
                                          start_date,
                                          end_date=None,
                                          room_id=None):
        query = Room.get_query(info)
        check_room = query.filter(RoomModel.id == room_id).first()
        ratio = RoomAnalyticsRatios.get_analytics_ratios_per_room(
            self, query, start_date, end_date, room_id)
        if not check_room:
            raise GraphQLError("Room not found")
        return RatiosPerRoom(ratios=[], ratio=ratio)
```
Also remember to allow the non-admins to view these analytics like @RachaelNantale  pointed out.

Reply:
Nice one @nakatuddesuzan 
1. Please, can you be specific on what is not working well on develop?
2. I don't think it's an efficient approach to always query the room model even when room id hasn't been provided.
3. This implementation does not handle a case where room id hasn't been provided.

Reply:
<img width="1087" alt="Screen Shot 2019-04-01 at 23 28 07" src="https://user-images.githubusercontent.com/39129473/55357530-e055d700-54d5-11e9-9429-b8ee6bcb2107.png">
This is what happens when this query is run while on develop. Kindly try it on your side to confirm. It could be different on your end

Reply:
About the case where the room_id is not provided, So like i pointed out that i went throgh the code. I realised thus, That very function you are working with is meant for for a specific room, 
```
@Auth.user_roles('Admin', 'Default User')
    def resolve_analytics_ratios(
            self, info, start_date, end_date=None):
        query = Room.get_query(info)
        ratio = RoomAnalyticsRatios.get_analytics_ratios(
            self, query, start_date, end_date)
        return ratio
```
That one pasted above is the one menat for all rooms(for when we don't specify the room_id) and below is the fixture, kindly run this query below to confirm. 
```
query {
    analyticsRatios(startDate:"Jul 10 2018", endDate:"Jul 29 2018"){
        bookings
        checkins
        cancellations
        cancellationsPercentage
        checkinsPercentage
        appBookings
        appBookingsPercentage
}
}
```

Reply:
okay, I had gotten it wrong.   I now get it, I guess it will be easier if you include both queries for when a room_id is provided and when room_id is not provided or screenshots so that the difference comes out well for a reviewer.

Reply:
Sorry , I had gotten that bit wrong, i was running the smae query for both instances. I think everything is okay. Just try to remove that second if statement. `if room_id:` by doing something like this if possible
```
def resolve_analytics_ratios_per_room(
            self, info, start_date, end_date=None, room_id=None):
        query = Room.get_query(info)
        ratio = RoomAnalyticsRatios.get_analytics_ratios_per_room(
            self, query, start_date, end_date, room_id) 
        if room_id:
            check_room = query.filter(RoomModel.id == room_id).first()
            if not check_room:
                raise GraphQLError("Room not found")
            return RatiosPerRoom(ratios=[], ratio=ratio)    
        return RatiosPerRoom(ratios=ratio, ratio={})
```